runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -1367,6 +1367,11 @@ bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
          case TR::java_lang_Math_max_L:
          case TR::java_lang_Math_min_L:
             return !comp()->getOption(TR_DisableMaxMinOptimization);
+         case TR::java_lang_Math_max_F:
+         case TR::java_lang_Math_min_F:
+         case TR::java_lang_Math_max_D:
+         case TR::java_lang_Math_min_D:
+            return !comp()->getOption(TR_DisableMaxMinOptimization) && cg()->getSupportsInlineMath_MaxMin_FD();
```

Comment:
@r30shah I confirmed that with this change the min/max calls will only be replaced on Z (so no crash on x86), but that means that other platforms that do define the fmin/fmax/dmin/dmax opcodes will still not use them. 

Is this a separate issue, or should the change for the other platforms go in this PR?

Reply:
I am assuming, you tried your test to see if it handles all the case on x86 platform, can you share what was your observation? i.e, which test case failed with inlining floating point min and max on x86 or any other platform you saw this failing ? 

I think we can create separate issue for that, but like to know specifics first.

Reply:
I get this without the change
```
Assertion failed at /home/jenkins/workspace/Build_JDK11_x86-64_linux_Personal/omr/compiler/codegen/OMRTreeEvaluator.cpp:584: false
VMState: 0x0005ff04
	Opcode fmin is not implemented
```

Reply:
Ok I am ok with opening up separate issue for X not supporting max and min, FYI @hzongaro. 

@matthewhall2  regarding to other platform, does your unit test pass (On P and aarch64) ?

Reply:
It passed on P.

However looking at aarch64, it appears that the quiet bit may be automatically set by the cpu. It failed some (but not all) of the tests with `-Xjit:count=1` (and count=0) set ([aarch64 test](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk11_j9_sanity.functional_aarch64_linux_Personal/612/)).

Reply:
@matthewhall2 my recommendation (Based on what you said) is to enable this on Z and P. And have issue opened up in OpenJ9 to checkout other platforms. 

FYI @zl-wang / @IBMJimmyk  , this change would enable the code for floating point max and min operations on P. So far, the corner cases Matthew has in this PR works (That checks for special cases like +0/-0 and NaNs). Would appreciate if you can confirm if it is OK for P or not. 

For AArch64 - tagging @0xdaryl.

Reply:
@matthewhall2 @r30shah   i have this item in P's backlog:  https://github.ibm.com/runtimes/openj9-jit-power/issues/416
will this PR end up with code of comparable performance?

Reply:
@r30shah looks like my test didn't inline. It fails on P with inlining enabled

Reply:
Ok, We should add P as well to the issue for fixing and enabling min/maxFD on other platform as recommended in https://github.com/eclipse-openj9/openj9/pull/20185#discussion_r1797047516 So please open an issue for that. 

The change in its current state (after you addressed concern and added the check for Code-Gen Flag, It should only be inlining it on Z. So we should be good.